third_party_libraries/kml.js

```diff
@@ -212,9 +213,10 @@ L.Util.extend(L.KML, {
 		el = xml.getElementsByTagName('Placemark');
 		for (var j = 0; j < el.length; j++) {
 			if (!this._check_folder(el[j], xml)) { continue; }
-			l = this.parsePlacemark(el[j], xml, style);
-
-			if (l) { layers.push(l); }
+			var parsedLayers = this.parsePlacemark(el[j], xml, style);
```

Comment:
They are in fact the same.There are two different paths the code may take that builds a set of these leaflet marker layers. I'm not sure what triggers what path it goes through but I've only ever been able to get it to step through the `parseFolder` function. The code I replaced was also duplicated across both parts, I just updated it to my new way of creating layers 

I